packages/gestalt/src/ComboBox.js

```diff
@@ -11,59 +11,132 @@ import {
   useMemo,
   useRef,
   useState,
+  type Ref,
 } from 'react';
 import Box from './Box.js';
 import Layer from './Layer.js';
 import Popover from './Popover.js';
 import Text from './Text.js';
 import InternalTextField from './InternalTextField.js';
 import Tag from './Tag.js';
-import ComboBoxItem, { type ComboBoxItemType } from './ComboBoxItem.js';
+import ComboBoxItem from './ComboBoxItem.js';
 import { ESCAPE, TAB, ENTER, UP_ARROW, DOWN_ARROW } from './keyCodes.js';
 import handleContainerScrolling, {
   KEYS,
   type DirectionOptionType,
 } from './utils/keyboardNavigation.js';
 
 type Size = 'md' | 'lg';
+type OptionType = {|
+  label: string,
+  subtext?: string,
+  value: string,
+|};
 
 type Props = {|
-  // REQUIRED
+  /**
+   * Label to describe the clear button's purpose.
+   */
   accessibilityClearButtonLabel: string,
+  /**
+   * Unique id to identify each ComboBox. Used for [accessibility](https://gestalt.pinterest.systems/combobox#Accessibility) purposes.
+   */
   id: string,
+  /**
+   * Provide a label to identify the ComboBox field.
+   */
   label: string,
-  options: $ReadOnlyArray<ComboBoxItemType>,
+  /**
+   * The data for each selection option. See [subtext](https://gestalt.pinterest.systems/combobox#With-subtext) variant to learn more.
+   */
+  options: $ReadOnlyArray<{|
```

Comment:
It looks like this is the OptionType above? You can use that type here and docgen will show the correct type in the docs. That only breaks when importing types from other files or when types are heavily nested.

Reply:
It doesn't! It only picks direct usages. As soon as you do Array etc if only shows the name! It took me a while to figure out.

Reply:
![image](https://user-images.githubusercontent.com/10593890/152012338-332abf38-1037-4ad8-b48b-76d54c80cd49.png)

Reply:
Dang, I thought it took one more level of depth for docgen to lose the type. Womp womp.